src/main/java/UseCase/RegisterInteractor.java

```diff
@@ -0,0 +1,46 @@
+package UseCase;
+import Entities.User;
+import OutputBoundary.loginOutputBoundary;
+import OutputData.loginOutputData;
+import InputData.loginInputData;
+import InputBoundary.RegisterInputBoundary;
+import Database.userAccessInterface;
+import Database.userDsData;
+import Entities.userFactory;
+
+public class RegisterInteractor implements RegisterInputBoundary{
+    private final loginOutputBoundary registerPresenter;
+    private final userFactory userFactory;
+
+    userAccessInterface Allusers;
+
+    public RegisterInteractor(loginOutputBoundary registerPresenter, userAccessInterface Allusers, userFactory userFactory){
+        this.Allusers = Allusers;
+        this.registerPresenter =registerPresenter;
+        this.userFactory = userFactory;
+    }
+    @Override
+    public void register(loginInputData loginID){
+        if (Allusers.exists(loginID.getUsername())){
+            //TODO: prepare user exists failure view
+        } else if (loginID.getPassword() == "") {
```

Comment:
The username field could also be empty.

Reply:
Ah yes I will add that in, thanks

Reply:
> If the password field can be left empty, I presume the username field could also be left empty.
> 
> Just an aside, are we having minimum characters for each? I guess it would add an unnecessary layer of complexity at the "prototype" stage.

lol i did not read the whole comment. the fail view is presented if username/password is empty so no they can't be empty. Also as of now we don't have the minimum characters, we can add that in later if u want?

Reply:
What I had meant was that, initially, having no username entered but a password entered would have passed a success view. Probably should have been more concise.